### PR TITLE
Weather App

### PR DESCRIPTION
**Pull request by Chaseknis**
- Hit openwether.org API
- Run linters for errors
- Update readme.md
- use react components
- Update Redux directory
- update routers directory

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,228 @@
...
+
+  - [Yes]()
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (Davidosky007) [inline README.md]: * Kindly fix the broken MIT link

File: src/__test__/details.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  it('should render the weather data when weather is not null', () => {
+    const mockWeather = {
+      name: 'New York',
+      country: 'US',
+      iconURL: 'https://someicon.com',
+      description: 'sunny',
+      temp: 20,
+      temp_min: 15,
+      temp_max: 25,
```
Comment (Davidosky007) [inline src/__test__/details.test.js]: * Some tests are failing, Kindly fix them 
image

File: src/index.js [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
+import './index.css';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+  <BrowserRouter>
+    <Provider store={store}>
+      <React.StrictMode>
```
Comment (Davidosky007) [inline src/index.js:13]: * When the page loads, the web app should show the list of categories that could be filtered out by some parameter, for example by the category name. Kindly enable this feature as its a [requirement ](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=When%20the%20page%20loads%2C%20the%20webapp%20shows%20the%20list%20of%20categories%20that%20could%20be%20filtered%20out%20by%20some%20parameter%2C%20for%20example%20by%20the%20category%20name.)

File: src/components/.header.js.icloud
Comment (Davidosky007) [inline src/components/.header.js.icloud:1]: * Kindly use the right file extension for your files as i can't access your codes in here

File: src/components/.country.jsx.icloud
Comment (Davidosky007) [inline src/components/.country.jsx.icloud:1]: * Kindly use the right file extension for your files as i can't access your codes in here

File: src/components/.continent.jsx.icloud
Comment (Davidosky007) [inline src/components/.continent.jsx.icloud:1]: * Kindly use the right file extension for your files as i can't access your codes in here

File: src/index.css [hunk 4]
```diff
@@ -1,13 +1,29 @@
...
   margin: 0;
-  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
-    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
+  padding: 0;
+  font-family:
+    -apple-system,
+    BlinkMacSystemFont,
+    'Segoe UI',
+    'Roboto',
+    'Oxygen',
```
Comment (Davidosky007) [inline src/index.css]: * Kindly stick to the design guidelines as its a [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Typography%3A%20font%20face%2C%20size%20and%20weight.) Typography: font face, size, and weight.

File: src/redux/cities/citiesSlice.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import getWeatherData from '../api/api';
+
+export const fetchCities = createAsyncThunk(
+  'cities/fetchCities',
+  async (cityNames) => {
+    // console.log(cityNames);
```
Comment (Davidosky007) [inline src/redux/cities/citiesSlice.js]: * Kindly remove all commented code from your codebase as it makes your code bulky and unprofessional

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Chaseknis 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 

## Required Changes ♻️
* Kindly highlight some interesting piece of code or something you built that you are very proud of in your video presentation 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (Whoistolu) [reply README.md]: - Well resolved 👍🏽

File: src/__test__/details.test.js (hunk 2 above)
Comment (Whoistolu) [reply src/__test__/details.test.js]: - [ ] [REQUIRED] Hello @Chaseknis , there are still failing tests. Let's resolve this.
Screenshot 2023-03-18 at 18 43 10

Screenshot 2023-03-18 at 19 03 04

File: src/index.js (hunk 3 above)
Comment (Whoistolu) [reply src/index.js:13]: - Well resolved 👍🏽

File: src/components/.header.js.icloud
Comment (Whoistolu) [reply src/components/.header.js.icloud:1]: - Well resolved 👍🏽

File: src/components/.country.jsx.icloud
Comment (Whoistolu) [reply src/components/.country.jsx.icloud:1]: - Well resolved 👍🏽

File: src/components/.continent.jsx.icloud
Comment (Whoistolu) [reply src/components/.continent.jsx.icloud:1]: - Well resolved 👍🏽

File: src/index.css (hunk 4 above)
Comment (Whoistolu) [reply src/index.css]: - Well resolved 👍🏽

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Chaseknis 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [x]  _Check the inline comments under the review for sections to improve._
- [x] [REQUIRED] As stated by the previous reviewer, kindly highlight some interesting piece of code or something you built that you are very proud of in your video presentation

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @Chaseknis,

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by Chaseknis** [comment]
thank so much @EricMbouwe

**Pull request merged**